creators_custom/TripsCreatorFenix.py

```diff
@@ -0,0 +1,173 @@
+        service_sunday.SetStartDate(self.config['feed_info']['start_date'])
+        service_sunday.SetEndDate(self.config['feed_info']['end_date'])
+        service_sunday.SetWeekdayService(False)
+        service_sunday.SetWeekendService(False)
+        service_sunday.SetDayOfWeekHasService(6, True)
+
+        # add trips for all routes
+        for route_ref, route in sorted(routes.iteritems()):
+            line = schedule.AddRoute(
+                short_name=route.ref,
+                long_name=route.name,
+                route_type="Bus")
+            line.agency_id = agency.agency_id
+            line.route_desc = "TEST DESCRIPTION"
+            line.route_url = "http://www.consorciofenix.com.br/horarios?q=" + str(route.ref)
+            line.route_color = "1779c2"
+            line.route_text_color = "ffffff"
+
+            weekday = {}
+            saturday = {}
+            sunday = {}
+
+            for day in route.horarios:
+                sday = day.encode('utf-8')
+
+                if sday.startswith(WEEKDAY):
+                    weekday[sday.replace(WEEKDAY + ' - Saída ', '')] = route.horarios[day]
+                elif sday.startswith(SATURDAY):
+                    saturday[sday.replace(SATURDAY + ' - Saída ', '')] = route.horarios[day]
+                elif sday.startswith(SUNDAY):
+                    sunday[sday.replace(SUNDAY + ' - Saída ', '')] = route.horarios[day]
+                else:
+                    raise RuntimeError("Unknown day in Fenix data: " + day)
+
+            self.add_trips_by_day(schedule, line, service_weekday, route, weekday, WEEKDAY)
+            self.add_trips_by_day(schedule, line, service_saturday, route, saturday, SATURDAY)
+            self.add_trips_by_day(schedule, line, service_sunday, route, sunday, SUNDAY)
+
+    def add_trips_by_day(self, schedule, line, service, route, horarios, day):
+        # check if we even have service
+        if horarios is None or len(horarios) == 0:
+            return
+
+        if isinstance(route, RouteMaster):
+            # recurse into "Ida" and "Volta" routes
+            for sub_route in route.routes.values():
+                self.add_trips_by_day(schedule, line, service, sub_route, horarios, day)
+            return
+
+        # have at least two stops
+        if len(route.stops) < 2:
+            sys.stderr.write("Skipping Route, has no stops: " + str(route) + "\n")
+            return
+
+        # check if we have a match for the first stop
+        key = route.match_first_stops(horarios.keys())
+
+        if key is None:
+            # Do not print debug output here, because already done in route.match_first_stops()
+            return
+
+        if route.ref == DEBUG_ROUTE:
+            print "\n\n\n" + str(route)
+            print day + " - " + key
+
+        # get shape id
+        shape_id = str(route.id)
+        try:
+            schedule.GetShape(shape_id)
+        except KeyError:
+            shape = transitfeed.Shape(shape_id)
```

Comment:
unresolved `transitfeed`
These exist a couple of times, but I guess fixing it once will fix it for all.